contracts/src/IVerifier.sol

```diff
@@ -47,100 +52,159 @@ interface IVerifier {
+     * @param _offset Starting index of array of models to be fetched
+     * @param _limit Number of models to be fetched
      * @return models List of models
      */
     function getModelsByOwnerAddress(
-        address ownerAddress,
-        uint32 offset,
-        uint32 limit
-    ) external view returns (ModelArrayElement[] memory models);
+        address _ownerAddress,
+        uint32 _offset,
+        uint32 _limit
+    ) external view returns (ModelArrayElement[] memory);
 
     /**
      * @notice Update info of the requested model.
      * @dev 'contentId' & 'ownerAddress' cannot be updated.
-     * @param modelContentId Hash (content ID / address of IPFS) of model
-     * @param modelName Updated name of model
-     * @param modelDescription Updated description of model
+     * @param _modelContentId Hash (content ID / address of IPFS) of model
+     * @param _modelName Updated name of model
+     * @param _modelDescription Updated description of model
      */
     function updateModel(
-        Hash modelContentId,
-        string calldata modelName,
-        string calldata modelDescription
+        Hash _modelContentId,
+        string calldata _modelName,
+        string calldata _modelDescription
     ) external;
 
     /**
      * @notice Disable the requested model.
      * @dev Model is logically disabled.
-     * @param modelContentId Hash (content ID / address of IPFS) of model
+     * @param _modelContentId Hash (content ID / address of IPFS) of model
      */
-    function disableModel(Hash modelContentId) external;
+    function disableModel(Hash _modelContentId) external;
 
     /**
-     * @notice Receives and stores commitments of testing results. Generates a random challenge in return.
-     * @dev Need to consider setting thresholds when generating a random challenge. Fine-tune them accordingly.
-     * @param commitmentModel Hash value of URL of targeted model
-     * @param commitmentData Hash value of testing data
-     * @param commitmentResults Root hash value of Merkle tree of testing results
-     * @return commitmentId commitment ID (need to provide when calling 'reveal()' & 'verify()')
-     * @return challenge Random challenge
+     * @notice Store commit of testing results, and generate a challenge in return.
+     * @dev 'commitId' is generated from '_modelContentId', '_merkleRoot' and sender's address.
+     * @param _modelContentId Hash (content ID / address of IPFS) of model
+     * @param _merkleRoot Root hash of Merkle tree of testing results
+     * @return commitId Commit ID
+     * @return challenge Challenge
+     * @return difficulty Difficulty of challenge
      */
     function commit(
-        bytes32 commitmentModel,
-        bytes32 commitmentData,
-        bytes32 commitmentResults
-    ) external returns (uint256 commitmentId, bytes32 challenge);
+        Hash _modelContentId,
+        Hash _merkleRoot
+    ) external returns (Hash, Hash, uint8);
```

Comment:
There is a ERC for commitment contract. Please refer to it.

https://eips.ethereum.org/EIPS/eip-5732